Unit tests for contract proxies

pcppcp opened the pull request:
- [x] secret registry
- [x] token networks registry

Fixes #1541

palango commented on raiden/network/proxies/secret_registry.py:
```diff
@@ -2,13 +2,18 @@
+    is_binary_address,
+    to_normalized_address,
+    event_abi_to_log_topic,
+    encode_hex
```
pls add trailing commas, there's multiple places

palango commented on raiden/network/proxies/token_network_registry.py:
```diff
@@ -70,18 +73,19 @@ def __init__(
         self.token_to_tokennetwork = dict()
 
-    def token_to_tokennetwork(self, token_address: typing.TokenAddress):
+    def get_token_network(self, token_address: typing.TokenAddress):
```
Pls add a return type annotation as well.

palango commented on raiden/network/rpc/client.py:
```diff
@@ -320,8 +321,8 @@ def deploy_solidity_contract(
             all_contracts,
-            libraries,
-            constructor_parameters,
+            libraries=dict(),
```
This will be the same object all the time, I think we should make this `None` and create a new list if it's `None` in the body.

palango commented on raiden/tests/unit/contracts/fixtures/contracts.py:
```diff
@@ -0,0 +1,78 @@
+        return deploy_client.deploy_solidity_contract(
+            contract_name,
+            compiled,
+            dict(),
```
For readability keyword args would be nice

pcppcp commented:
@palango addressed your comments

> One issue is where to put the files: As the tests are blockchain based I think we should move it into the integration tests folder.

Yeah I was wondering where to put these - but since they are not "integrating" much - they only require blockchain as a backend*. Maybe we need a better name for these (component test?)

* (and since it's an interface being tested, `eth-tester` will be a good option to provide the backend in the future for these. ATM they're using geth like all the other tests.)

palango commented on raiden/network/proxies/token_network_registry.py:
```diff
@@ -70,18 +74,19 @@ def __init__(
         self.token_to_tokennetwork = dict()
 
-    def token_to_tokennetwork(self, token_address: typing.TokenAddress):
+    def get_token_network(self, token_address: typing.TokenAddress) -> Union[None, str]:
```
It's always nicer to write `Union[None, str]` as `Optional[str]`

And it should be `Optional[typing.Address]` here as the address gets converted to a bytestring.

palango commented:
> Yeah I was wondering where to put these - but since they are not "integrating" much - they only require blockchain as a backend*. Maybe we need a better name for these (component test?)

component tests sounds good to me. Until we have the `eth-tester` backend I think they should go into the integration folder. But lets ask @LefterisJP for his opinion

LefterisJP commented on raiden/network/proxies/token_network_registry.py:
```diff
@@ -1,19 +1,23 @@
     to_canonical_address,
+    encode_hex,
+    event_abi_to_log_topic,
+    to_checksum_address
```
trailing comma here ;)

LefterisJP commented on raiden/tests/unit/contracts/fixtures/contracts.py:
```diff
@@ -0,0 +1,77 @@
+import pytest
+from raiden.network.proxies import (
+    SecretRegistry,
+    TokenNetworkRegistry
```
trailing comma

LefterisJP commented on raiden/tests/unit/contracts/fixtures/contracts.py:
```diff
@@ -0,0 +1,77 @@
+from raiden_contracts.constants import (
+    CONTRACT_TOKEN_NETWORK_REGISTRY,
+    CONTRACT_SECRET_REGISTRY,
+    CONTRACT_HUMAN_STANDARD_TOKEN
```
here too

LefterisJP commented on raiden/tests/unit/contracts/test_network_registry.py:
```diff
@@ -0,0 +1,52 @@
+    TransactionThrew,
+    InvalidAddress,
+    AddressWithoutCode,
+    NoTokenManager
```
here too

LefterisJP commented:
yeah since they use geth for backend they should go to the integration directory (we group them all so we can run them in the same job in Travis too)

pcppcp commented:
@LefterisJP I've fixed damn commas and the tests reside in the `integration` folder now.
The contract tests runs together with `long_running` tests, in order to save some time during CI runs.

hackaugusto commented on raiden/tests/fixtures/__init__.py:
```diff
@@ -0,0 +1,12 @@
+
+
+@pytest.fixture
+def get_random_bytes():
```
this is really silly, there is no reason to use a fixture for this, just add the standalone function in the tests utils

hackaugusto commented on raiden/network/proxies/token_network_registry.py:
```diff
@@ -70,18 +74,19 @@ def __init__(
+        address = to_canonical_address(address)
 
-        if address == b'':
+        if address == b'\x00' * 20:
```
please, define this in the constant file and give it a name, e.g. NULL_ADDRESS

hackaugusto commented on raiden/tests/integration/contracts/test_network_registry.py:
```diff
@@ -0,0 +1,52 @@
+
+@pytest.fixture
+def get_random_address(get_random_bytes):
+    return lambda: get_random_bytes(20)
```
again, we don't need this pattern. please use the existing utilities:

https://github.com/raiden-network/raiden/blob/990827def9f24266ee01287940485e48aea04bf6/raiden/tests/utils/factories.py#L80-L81
